Lib/ufo2ft/featureWriters/gdefFeatureWriter.py

```diff
@@ -73,29 +78,34 @@ def _sortedGlyphClass(self, glyphNames):
         return sorted(n for n in self.context.orderedGlyphSet if n in glyphNames)
 
     def _makeGDEF(self):
-        fea = ast.TableBlock("GDEF")
-        categories = self.context.openTypeCategories
+        gdefTableBlock = self.context.gdefTableBlock
+        if not gdefTableBlock:
+            gdefTableBlock = ast.TableBlock("GDEF")
 
         if "GlyphClassDefs" in self.context.todo:
+            categories = self.context.openTypeCategories
             glyphClassDefs = ast.GlyphClassDefStatement(
                 ast.GlyphClass(self._sortedGlyphClass(categories.base)),
                 ast.GlyphClass(self._sortedGlyphClass(categories.mark)),
                 ast.GlyphClass(self._sortedGlyphClass(categories.ligature)),
                 ast.GlyphClass(self._sortedGlyphClass(categories.component)),
             )
-            fea.statements.append(glyphClassDefs)
+            gdefTableBlock.statements.append(glyphClassDefs)
 
         if "LigatureCarets" in self.context.todo:
             ligatureCarets = [
                 ast.LigatureCaretByPosStatement(ast.GlyphName(glyphName), carets)
                 for glyphName, carets in self.context.ligatureCarets.items()
             ]
-            fea.statements.extend(ligatureCarets)
+            gdefTableBlock.statements.extend(ligatureCarets)
 
-        return fea
+        return gdefTableBlock
 
     def _write(self):
         feaFile = self.context.feaFile
-        feaFile.statements.append(self._makeGDEF())
+        if self.context.gdefTableBlock:
+            self._makeGDEF()
+        else:
+            feaFile.statements.append(self._makeGDEF())
```

Comment:
this is a bit weird, that depending on external cirucmstances you use or not the return value of `self._makeGDEF`. Maybe internalize that logic inside `_makeGDEF` itself, or even better move the whole content of` _makeGDEF` to this `_write` method which is short enough - that's what this writer does, it makes a GDEF.

Reply:
OK. `_makeGDEF()` has been merged into `_write()`.